plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Products.php

```diff
@@ -216,4 +225,19 @@ private static function count_user_products() {
 
 		return $products_query->found_posts;
 	}
+
+	public function maybe_redirect_to_add_product_tasklist() {
+		$screen = get_current_screen();
+		if ( 'edit' === $screen->base && 'product' === $screen->post_type ) {
+			// wp_count_posts is cached.
+			$counts = (array) wp_count_posts( $screen->post_type );
+			unset( $counts['auto-draft'] );
+			$count = array_sum( $counts );
+			if ( $count > 0 ) {
```

Comment:
Note that this class also has the `has_products()` function, I was going to use this, but it unfortunately doesn't handle products that are trashed but not permanently deleted. Causing it to still redirect to the task if trashed products where present. Given there would be no other way to permanently delete the products without the table, I decided to use the above approach.
I used the same logic here, as the product admin list table class that displays the blank state: https://github.com/woocommerce/woocommerce/blob/ef015f740f89c35cebf34bc0c39a971be6723b68/plugins/woocommerce/includes/admin/list-tables/abstract-class-wc-admin-list-table.php#L65